packages/lit-html/src/lit-html.ts

```diff
@@ -705,14 +725,24 @@ export type Part =
 export class NodePart {
   readonly type = NODE_PART;
   _value: unknown;
-  protected _directive?: Directive;
+  /** @internal */
+  _directive?: Directive;
 
   constructor(
     private _startNode: ChildNode,
     private _endNode: ChildNode | null,
     public options: RenderOptions | undefined
   ) {}
 
+  get parentNode(): Node {
+    return this._startNode.parentNode!;
+  }
+
+  /** @internal */
+  _setEndNode(node: ChildNode | null) {
+    this._endNode = node;
+  }
```

Comment:
During hydration, we need to construct a `NodePart` before we've reached its end marker; for now I've added a `_setEndNode` method rather than make `_endNode` public; TBD

Reply:
Unless we do a rollup build for SSR (which we discussed in chat and decided to go the private-support approach for top-level exports), any mangled _fields_ of those members still hit the issue, and need to be in the _reserved_ list (not the cross-package mangle list). So they're both not hidden and not mangled.  It's just a choice we need to decide on and then be consistent.

Reply:
Sorry I take that back, _startNode/_endNode are only used in hydrate, which can be mangled, so I'll go for that.